### PR TITLE
Added ftpOverProxy

### PR DESCRIPTION
**Pull request by vives**

### CODE REVIEW

File: src/main/resources/file/fileftpoverproxy-template.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    <parameter name="ftpServer"/>
+    <parameter name="ftpPort"/>
+    <parameter name="ftpUsername"/>
+    <parameter name="ftpPassword"/>
+    <parameter name="keepAliveTimeout"/>
+    <parameter name="controlKeepAliveReplyTimeout"/>
+    <parameter name="targetPath"/>
+    <parameter name="targetFile"/>
+    <parameter name="binaryTransfer"/>
+    <parameter name="localActive"/>
```
Comment (RKathees) [inline src/main/resources/file/fileftpoverproxy-template.xml]: Add the description for the parameters.

File: src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * FTP Over Proxy
+ */
+public class FileFtpOverProxy extends AbstractConnector implements Connector {
+    private static final Log log = LogFactory.getLog(FileFtpOverProxy.class);
+
+    public void connect(MessageContext messageContext) {
```
Comment (pamod) [inline src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: javaDocs ?

File: src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+                ftpOverHttp(proxyHost, proxyPort, proxyUsername, proxyPassword, ftpServer, ftpPort,
+                        ftpUsername, ftpPassword, messageContext);
+        generateResult(messageContext, resultStatus);
+    }
+
+    /**
+     * Generate the result.
+     *
+     * @param messageContext The message context that is generated for processing the file
+     * @param resultStatus   true/false
```
Comment (pamod) [inline src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: Can you elaborate this more ? true/false in a boolean would be understandable from it's type

File: src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+        } catch (IOException e) {
+            //log.error("Error occurred while uploading:" + e.getMessage(), e);
+            handleException("Could not connect to FTP ftpServer: " + e.getMessage(), e, messageContext);
+        } finally {
+            if (ftp.isConnected()) {
+                try {
+                    ftp.disconnect();
+                    ftp.logout();
+                } catch (IOException f) {
+                    // do nothing
```
Comment (pamod) [inline src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: This should not be silently ignored IMO. Could you add a log ?

Comment (vives) [reply src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: Fixed in wso2-extensions/esb-connector-file#28

File: src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java (hunk 3 above)
Comment (vives) [reply src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: Fixed in wso2-extensions/esb-connector-file#28

File: src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+import org.wso2.carbon.connector.util.FileConstants;
+import org.wso2.carbon.connector.util.ResultPayloadCreate;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * FTP Over Proxy
```
Comment (Niranjan-K) [inline src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: JavaDoc issue. Same as mentioned in [1]

[1] - https://github.com/wso2-extensions/identity-outbound-auth-office365/pull/1

Comment (vives) [reply src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/28

File: src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java (hunk 2 above)
Comment (vives) [reply src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java]: Fixed in https://github.com/wso2-extensions/esb-connector-file/pull/28

**Pull request merged**